nixos/modules/services/networking/nats.nix

```diff
@@ -0,0 +1,174 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.nats;
+
+  confFile = pkgs.writeText "nats.conf" (builtins.toJSON cfg.settings);
+
+in {
+
+  ### Interface
+
+  options = {
+    services.nats = {
+      enable = mkEnableOption "NATS messaging system";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.nats-server;
+        defaultText = "pkgs.nats-server";
+        description = "The nats-server package to use";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "nats";
+        description = "User account under which NATS runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "nats";
+        description = "Group under which NATS runs.";
+      };
+
+      serverName = mkOption {
+        default = "nats";
+        example = "n1-c3";
+        type = types.str;
+        description = ''
+          Name of the NATS server, must be unique if clustered.
+        '';
+      };
+
+      jetstream = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable JetStream";
+      };
+
+      host = mkOption {
+        default = "127.0.0.1";
+        example = "0.0.0.0";
+        type = types.str;
+        description = ''
+          Host to listen on.
+        '';
+      };
+
+      port = mkOption {
+        default = 4222;
+        example = 4222;
+        type = types.int;
+        description = ''
+          Port on which to listen.
+        '';
+      };
+
+      dataDir = mkOption {
+        default = "/var/lib/nats";
+        type = types.path;
+        description = ''
+          The data directory.
+        '';
+      };
+
+      settings = mkOption {
+        default = { };
+        type = types.submodule {
+          freeformType = with lib.types; attrsOf anything;
+          options = { };
+        };
+        example = literalExample ''
+          {
+            jetstream = {
+              max_mem = "1G";
+              max_file = "10G";
+            };
+          };
+        '';
+        description = ''
+          Declarative Nats configuration. See https://docs.nats.io/nats-server/configuration
```

Comment:
:heart: clickable links.

Reply:
Don't we need to specially format it for the docs?

Reply:
Yes, that is what I meant... I love clickable links. Can you please make it clickable... Sorry, will be more clear next time 😄

Reply:
I understood, fixed it in the new commit :smile: